### PR TITLE
refactor: remove @angular/http

### PR DESCRIPTION
**Pull request by CaerusKaru**
BREAKING CHANGE:
* `in-memory-web-api` can no longer be used with `@angular/http`

### CODE REVIEW

**Comment by CaerusKaru** [comment]
cc @IgorMinar

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angular-in-memory-web-api",
-  "version": "0.7.0",
+  "version": "1.0.0",
```
Comment (IgorMinar) [inline package.json]: 0.8 please

**Comment by IgorMinar** [comment]
@wardbell FYI ^

**Comment by wardbell** [comment]
Looking forward to a review this week.

FWIW, if we’re really getting rid of old `Http`, it’s kind of nutty to keep the abstraction that supported both.  I mentioned this I think in our conversation. Much of the muddiness in this library is due to the now pointless support for both http libraries.

I’m particularly unhappy with the dance that lead to `RequestCore` and `HeadersCore` (see `interfaces.ts`). It made life difficult for developers to get at the body and headers in a type-ful way. I don't know if you cleaned this up in this PR.

Not saying we should hold the PR up for those reasons. Just commentary for the moment.

**Comment by IgorMinar** [comment]
@wardbell I'd prefer any major refactoring should be done separately from the removal of http because otherwise this PR will grow uncontrollably. Let's get this in and 0.8 out. sgty?

**Comment by wardbell** [comment]
@IgorMinar I'm with you on that. I would have done the same. I just hate that it's so twisted to accommodate a feature (old Http) it no longer supports, leaving the code in a mysterious, convoluted state that reflects poorly on the author (me).  I realize that this is my ego trip.

**Comment by IgorMinar** [comment]
@wardbell we can do that as a followup step and your contributions are welcome, but in the meantime this removal is blocking a lot of other work in angular/angular so we need to get moving on this.

I asked @brandonroberts to take over this PR and cut the 0.8 release with this change.

**Pull request closed without merging**